pupil-spa/src/app/services/login-error-diagnostics/login-error-diagnostics.service.ts

```diff
@@ -22,14 +22,15 @@ export class LoginErrorDiagnosticsService {
     if (err.status !== 0) {
       return;
     }
+    const isBrowserStatusOnline = this.window.navigator.onLine;
+    if (!isBrowserStatusOnline) {
+      this.loginErrorService.changeMessage('Internet disconnected');
+      return;
+    }
     const canAccessAuthURL = await this.canAccessURL(APP_CONFIG.authURL);
     if (!canAccessAuthURL) {
       this.loginErrorService.changeMessage(`Connection refused to ${APP_CONFIG.authURL}`);
```

Comment:
This logic is looking much better 💯 

"Connection refused" normally means the server refused the connection:

1. Nothing is listening on the IP:Port you are trying to connect to.
2. The port is blocked by a firewall.

Is that the right message here?

Reply:
@jon-shipley it's the exact message the browser is throwing on the console when the requested url is not accessible: `ERR_CONNECTION_REFUSED`
Perhaps better as `Requested authentication url <url> is inaccessible`?